### PR TITLE
Fix SDL not send choiceID with PerformInteraction

### PR DESCRIPTION
**Pull request by indlu**
Fixed SDL does not send choiceID with PerformInteraction

### CODE REVIEW

**Comment by indlu** [comment]
Please review @dcherniev, @okozlovlux , @LuxoftAKutsan

**Comment by LuxoftAKutsan** [comment]
@pold500 please review

**Comment by LuxoftAKutsan** [comment]
@indlu check buildservre

File: src/components/application_manager/include/application_manager/commands/mobile/perform_interaction_request.h [hunk 1]
```diff
@@ -45,7 +45,50 @@ namespace application_manager {
...
+    return value_;
+  }
+  bool is_initialized() const {
+    return value_state_;
+  }
 
+ private:
+  IntType value_;
+  bool value_state_;
+};
```
Comment (indlu) [inline src/components/application_manager/include/application_manager/commands/mobile/perform_interaction_request.h]: @pold500, fixed in 9d7244f

File: src/components/utils/include/utils/optional.h [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  }
+  OptionalImpl& operator=(const OptionalImpl& new_val) {
+    this->value_ = new_val.value_;
+    if (new_val.is_initialized()) {
+      this->value_state_ = new_val.value_state_;
+    }
+    return *this;
+  }
+  OptionalImpl& operator*() {
+    return *value_;
```
Comment (pold500) [inline src/components/utils/include/utils/optional.h]: My initial though that this method would return value_ and not Optional<Type>.
Besides that, I like such implementation much more.

Comment (indlu) [inline src/components/utils/include/utils/optional.h]: @pold500, fixed in 0abcbf6

**Comment by indlu** [comment]
Rebuild required

**Pull request merged**